File: solutions/elixir/anagram/5/lib/anagram.ex
defmodule Anagram do

  defp normalize(word) do 
      word
      |> String.downcase()
      |> String.replace(~r([^a-z]),"")
      |> String.to_charlist()
      |> Enum.sort()
  end
  @doc """
  Returns all candidates that are anagrams of, but not equal to, 'base'.

  """
  @spec match(String.t(), [String.t()]) :: [String.t()]
  def match(base, candidates) do
    normalized = normalize(base)
    Enum.filter(candidates,&(String.downcase(&1) != String.downcase(base) &&
                             normalize(&1)==normalized))
  end
end

File: solutions/clojure/anagram/5/src/anagram.clj
(ns anagram
   (:require [clojure.string :as str]))

(defn normalize [s] 
  (sort (str/lower-case s)))

(defn anagrams-for [word prospect-list] 
  (let [normalized-word (normalize word)]
   (->> prospect-list
        (filter #(and (not= (str/lower-case word)(str/lower-case %)) 
                      (= normalized-word (normalize %)))))))

